boto/resultset.py

```diff
@@ -19,10 +19,13 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
+from collections import Sequence
+from boto.vendored.six.moves import UserList
+
 from boto.s3.user import User
 
 
-class ResultSet(list):
+class ResultSet(UserList):
```

Comment:
Are you subclassing from `UserList` instead of `list` to satisfy some constraint of Jython?

My main concern here is that this is a subtle breaking change to the API because `UserList` is not a subclass of `list`. This has the unfortunate side effect of breaking any user that is using `isinstance` checks to see if a `ResultSet` is a `list`.